From 41ce80f61bab1e4cb78139bad4234fb595f15734 Mon Sep 17 00:00:00 2001
From: oldherl <oldherl@gmail.com>
Date: Tue, 25 Jun 2024 04:22:37 +1000
Subject: [PATCH] hwdb: fix keyboard of RedmiBook Pro 15 2022 (#33465)

Fix two problems of the keyboard of RedmiBook Pro 15 2022.
- Enter key in the main area was mapped to KP_Enter.
- When Fn is locked (to use F1-F12 without pressing Fn),
  Right Ctrl was mapped to Menu. Keeping it as Right Ctrl is more useful.

(cherry picked from commit 341f04fa330a1c893b80b12c44f91db4c35fb50a)

Related: RHEL-5950
---
 hwdb.d/60-keyboard.hwdb | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index 81ffc69fa1..48aa92b433 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -1753,6 +1753,15 @@ evdev:input:b0003v047FpC006*
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svn*:pn*:pvr*:rvnQuanta:rn30B7:rvr65.2B:*
  KEYBOARD_KEY_88=media                                  # "quick play
 
+###########################################################
+# Redmi
+###########################################################
+
+# RedmiBook Pro 15 2022
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnTIMI:pnRedmiBookPro152022:pvr*
+ KEYBOARD_KEY_9c=enter                                  # KP_enter in the main area is wrong
+ KEYBOARD_KEY_dd=rightctrl                              # Right Ctrl is preferrable over Menu
+
 ###########################################################
 # Samsung
 ###########################################################
